qiskit/transpiler/__init__.py

```diff
@@ -155,15 +228,599 @@
+
+Scheduling stage
+----------------
+
+.. seealso::
+    :ref:`transpiler-scheduling-description`
+        A guide-level explanation of scheduling concepts.
+
+The scheduling stage, if requested, is responsible for inserting explicit :class:`~.circuit.Delay`
+instructions to make idle periods of qubits explicit.  Plugins may optionally choose to do
+walltime-sensitive transformations, such as inserting dynamical decoupling sequences.
+
+The input to the scheduling stage is an ISA-compatible circuit.  The output of the scheduling stage
+must also be an ISA-compatible circuit, with explicit :class:`~.circuit.Delay` instructions that
+satisfy the hardware's timing information, if appropriate.
+
+The scheduling stage should set the ``node_start_time`` property in the pipeline's
+:class:`.PropertySet`.
+
+When writing :ref:`stage plugins <transpiler-preset-stage-plugins>`, the entry point for
+``scheduling`` is ``qiskit.transpiler.scheduling``.  The built-in plugins are:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Summary
+
+    * - :ref:`default <transpiler-preset-stage-scheduling-default>`
+      - Attempt to satisfy timing alignment constraints without otherwise scheduling.
+
+    * - :ref:`alap <transpiler-preset-stage-scheduling-alap>`
+      - Schedule the circuit, preferring operations to be as late as possible.
+
+    * - :ref:`asap <transpiler-preset-stage-scheduling-asap>`
+      - Schedule the circuit, preferring operations to be as soon as possible.
+
+.. _transpiler-preset-stage-scheduling-default:
+
+Built-in ``default`` plugin
+...........................
+
+Do nothing, unless the circuit already contains instructions with explicit timings.  If there are
+explicitly timed operations in the circuit, insert additional padding to ensure that these timings
+satisfy the alignment and other hardware constraints.
+
+.. _transpiler-preset-stage-scheduling-alap:
+
+Builtin ``alap`` plugin
+.......................
+
+Explicitly schedule all operations using an "as late as possible" strategy.  This uses the
+:class:`.ALAPScheduleAnalysis` algorithm to decide where to place gates.
+
+.. _transpiler-preset-stage-scheduling-asap:
+
+Builtin ``asap`` plugin
+.......................
+
+Explicitly schedule all operations using an "as soon as possible" strategy.  This uses the
+:class:`.ASAPScheduleAnalysis` algorithm to decide where to place gates.
+
+
+Custom pass managers
 ====================
 
 In addition to modifying preset pass managers, it is also possible to construct a pass
 manager to build an entirely custom pipeline for transforming input
 circuits. You can use the :class:`~.StagedPassManager` class directly to do
 this. You can define arbitrary stage names and populate them with a :class:`~.PassManager`
 instance. For example, the following code creates a new :class:`~.StagedPassManager`
-that has 2 stages, ``init`` and ``translation``.::
+that has two stages, ``init`` and ``translation``.
+
+.. code-block::
```

Comment:
I think we can make this code block testeable, seems complete:

```suggestion
.. plot::
    :include-source:
    :nofigs:
```